attestation-agent/attester/src/snp/mod.rs

```diff
@@ -44,6 +44,11 @@ impl Attester for SnpAttester {
             .get_ext_report(None, Some(data), Some(0))
             .context("Failed to get attestation report")?;
 
+        let certs = match certs {
+            Some(c) => c,
+            None => bail!("Certificate chain was not provided"),
+        };
```

Comment:
apart from the @fitzthum comment, this pattern matching can be rewritten to an if let

```suggestion
        let Some(certs) = certs else {
             bail!("Certificate chain was not provided");
        }
```